src/coreclr/src/vm/exceptionhandling.cpp

```diff
@@ -4664,8 +4663,26 @@ VOID DECLSPEC_NORETURN UnwindManagedExceptionPass1(PAL_SEHException& ex, CONTEXT
             controlPc = Thread::VirtualUnwindLeafCallFrame(frameContext);
         }
 
+        bool isManagedCode = ExecutionManager::IsManagedCode(controlPc);
```

Comment:
This condition does not look right to me. Why do we check `!isManagedCode` here?

Reply:
This detects the boundary when we've unwound from a managed to a native frame. If the NativeCallable method was the managed one, the native frame is its caller and that's when we need to fail. The codeInfo still has info on the managed frame.

Reply:
What happens in the case when the caller is managed code (unmanaged calli)? I think we should also fail with unhandled exception in this case.